### PR TITLE
Add JRuby language detection

### PR DESCRIPTION
**Pull request by robertjli**
### What does this PR do?

Add JRuby language detection via command line.

### Motivation

https://datadoghq.atlassian.net/browse/PROC-3233

### Describe how to test/QA your changes

Run a JRuby command (example below), and run the process agent and the core agent with language detection enabled.

```
process_config:
  process_collection:
    enabled: true
language_detection:
  enabled: true
```

Run `agent workload-list` and check that the JRuby process is correctly identified as Ruby.

##### JRuby example

[Install JRuby](https://github.com/jruby/jruby/wiki/JRubyDistributions) using your favorite package manager (`sudo apt-get install jruby`, for example), and create the following file:

test.rb
```ruby
sleep(3600)
```

Then run it in the background with `jruby test.rb &`. The pid should be printed to stdout.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: releasenotes/notes/jruby-detection-ff54fa80eaa0a70e.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+enhancements:
+  - |
+    Adds JRuby language detection
```
Comment (xornivore) [inline releasenotes/notes/jruby-detection-ff54fa80eaa0a70e.yaml:11]: May want to expand this description and provide more context for the change. Also can drop the top commented out block.

File: pkg/languagedetection/cli_detector.go [hunk 2]
```diff
@@ -71,10 +79,29 @@ func languageNameFromCommand(command string) languagemodels.LanguageName {
...
+// DetectLanguage uses a combination of commandline parsing and binary analysis to detect a
+// process's language
 func DetectLanguage(procs []languagemodels.Process) []*languagemodels.Language {
 	langs := make([]*languagemodels.Language, len(procs))
 	for i, proc := range procs {
+		// Language-specific detectors should precede matches on the command/exe
+		for _, detector := range cliDetectors {
+			lang, err := detector.DetectLanguage(proc)
+			if err != nil {
+				log.Warnf("unable to detect language for proc %d: %s", proc.GetPid(), err)
```
Comment (xornivore) [inline pkg/languagedetection/cli_detector.go:91]:
```suggestion
				log.Warnf("unable to detect language for process %d: %s", proc.GetPid(), err)
```

**Review by xornivore — COMMENTED** [review]
For folks unfamiliar with JRuby, do you mind adding some instructions on how to install and test it, maybe an example of an app?

File: pkg/languagedetection/cli_detector.go [hunk 3]
```diff
@@ -9,9 +9,15 @@ import (
 	"regexp"
 	"strings"
 
+	"github.com/DataDog/datadog-agent/pkg/languagedetection/internal/detectors"
 	"github.com/DataDog/datadog-agent/pkg/languagedetection/languagemodels"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
 )
 
+var cliDetectors = []languagemodels.Detector{
+	&detectors.JRubyDetector{},
```
Comment (just-chillin) [inline pkg/languagedetection/cli_detector.go:18]: Nit: Should this be a pointer? The recievers that implement languagemodels.Detector don't take a pointer type.

**Review by just-chillin — APPROVED** [review]
Lgtm once @xornivore's comments are addressed

**Comment by robertjli** [comment]
> For folks unfamiliar with JRuby, do you mind adding some instructions on how to install and test it, maybe an example of an app?

@xornivore I added instructions and example to the PR description

File: pkg/languagedetection/cli_detector.go (hunk 3 above)
Comment (robertjli) [reply pkg/languagedetection/cli_detector.go:18]: Nice catch, I missed this one.

**Review by hestonhoffman — APPROVED** [review]
Small edit

File: releasenotes/notes/jruby-detection-ff54fa80eaa0a70e.yaml (hunk 1 above)
Comment (robertjli) [reply releasenotes/notes/jruby-detection-ff54fa80eaa0a70e.yaml]: Done

**Pull request merged**